controllers/machineset_controller.go

```diff
@@ -384,6 +384,9 @@ func (r *MachineSetReconciler) getNewMachine(machineSet *clusterv1.MachineSet) *
 	machine.ObjectMeta.OwnerReferences = []metav1.OwnerReference{*metav1.NewControllerRef(machineSet, machineSetKind)}
 	machine.Namespace = machineSet.Namespace
 	machine.Spec.ClusterName = machineSet.Spec.ClusterName
+
+	machine.Labels[clusterv1.MachineSetLabelName] = machineSet.Name
```

Comment:
Should we check and initialize Labels, in case it's nil?

Reply:
added, thanks!